File: Python/arrays/conclusions/squareSortedArr.py
class Solution:
        def __init__(self):
                self.arr = [-4,-1,0,3,10]
                # self.data = int(input("Please provide the length of the array: "))
                # self.addToList(0)
        
        def squaredSortedArr(self):
                self.i = 0
                self.flag = False
                while(self.i<len(self.arr)):
                        if (self.flag == False):
                                self.arr[self.i] = pow(self.arr[self.i], 2)
                                if (self.i+1 == len(self.arr)):
                                        self.flag = True
                                        self.i = 0
                                else:
                                        self.i += 1
                        else:
                                if (self.i+1 != len(self.arr) and self.i+1 < len(self.arr)):
                                        if (self.arr[self.i] < self.arr[self.i+1]):
                                                self.temp = self.arr[self.i]
                                                self.arr[self.i] = self.arr[self.i+1]
                                                self.arr[self.i+1] = self.temp
                                                self.i = 0
                                                print(self.temp)
                                        else:
                                                self.i += 1
                                else:
                                        self.i += 1
                return self.arr
p1 = Solution()
print(p1.squaredSortedArr())

File: Javascript/arrays/squareSortedArr.js
var squaredElements = function(elements) {
    let self = this;
    self.i = 0;
    self.flag = false;
    while(self.i<elements.length) {
        if (!self.flag) {
            elements[self.i] = Math.pow(elements[self.i], 2);
            self.i++;
            if (self.i+1 == elements.length) {
                self.flag = true;
                self.i = 0;
            }
        } else {
            if (elements[self.i] < elements[self.i+1] && elements[self.i+1] !== undefined) {
                self.temp = elements[self.i]; 
                elements[i] = elements[self.i+1]
                elements[i+1] = self.temp;
                self.i = 0;
            } else {
                self.i++;
            }
        }
    }
    return elements;
}

console.log(squaredElements([-4,-1,0,3,10]))